Yoast/wordpress-seo
# Deprecate class-yoast-modal.php

## Summary

This PR can be summarized in the following changelog entry:

* Deprecates the Yoast_Modal class.

## Relevant technical choices:

* When looking into https://github.com/Yoast/wordpress-seo/pull/11504/files, I noticed the Yoast_Modal class is no longer being used. More specificly: the class was used in 2 modal configs (`class-metabox-keyword-synonyms-config.php` and `class-metabox-multiple-keywords-config.php`), but those configs were no longer used, because the modals have been replaced by react modals. 

## Test instructions

This PR can be tested by following these steps:

* In the metabox/sidebar click '+ Add synonyms' and '+ Add additional keyphrase'. In both cases, you should get a functional (React) modal. 

## Quality assurance
